IoTDeviceSimulator/web-client/src/DeviceTwin.js

```diff
@@ -146,76 +202,402 @@ export function DeviceTwin(props) {
+                }
+                else if (JSON.parse(deviceTwin.signalArray[i])["Behaviour"] === "Square") {
+                    for (let j = 0; j < len; j++) {
+                        let currData = parseFloat(JSON.parse(deviceTwin.signalArray[i])["Amplitude"]) * Math.sign(Math.sin((2 * Math.PI * (j * (parseFloat(deviceTwin.telemetrySendInterval) / 1000))) / (parseFloat(JSON.parse(deviceTwin.signalArray[i])["Wave Period"]) / 1000)));
+                        currDataArray[j] += currData;
+                    }
+                }
+            }
+            else {
+                for (let j = 0; j < len; j++) {
+                    currDataArray[j] += 0;
+                }
+            }
+        }
+    }
+
+    if (deviceTwin.valueIsBool) {
+        currDataArray = [];
+        for (let i = 0; i < len; i++) {
+            let currData = Math.round(Math.random());
+            currDataArray.push(currData);
+        }
+    }
+
+    const sineInfo = "A sinusoidal wave is a mathematical curve defined in terms of the sine trigonometric function, of which it is the graph.";
+    const constantInfo = "A constant wave is a wave that is same everywhere & having the same value of range for different values of the domain.";
+    const increasingInfo = "A stricty increasing wave is a wave in which Y-value increases with the increasing value on X-axis.";
+    const noiseInfo = "A noise wave is an unwanted random disturbance of a signal.";
+    const triangularInfo = "A triangular wave or triangle wave is a non-sinusoidal waveform named for its triangular shape.";
+    const sawtoothInfo = "The sawtooth wave is a kind of non-sinusoidal waveform named for its resemblance to the teeth of a plain-toothed saw with a zero rake angle.";
+    const squareInfo = "A square wave is a non-sinusoidal periodic wave in which the amplitude alternates at a steady frequency between fixed min & max values, with the same duration at min & max.";
+
+    const getContent = () => {
+        if (deviceTwin.signalArray[index] === "") {
+            return (<div className="behaviour-func">
+                <div className="behaviour-list">
+                    <Select value={behaviour} placeholder={"Select Behaviour"} options={options} onChange={changeBehaviour} style={{ width: "300px" }} itemRenderer={(option) => (
+                        < MenuItem><Tooltip placement="right" content={option.value === 'Sine' ? sineInfo : option.value === 'Constant' ? constantInfo : option.value === 'Linear' ? increasingInfo : option.value === 'Noise' ? noiseInfo : option.value === 'Triangular' ? triangularInfo : option.value === 'Sawtooth' ? sawtoothInfo : squareInfo}><div className="option">{option.label}</div></Tooltip></MenuItem>
+                    )} >
+                    </Select>
+                </div>
+                <div className="behaviour-prop">
+                    {
+                        behaviour === "Sine" ?
+                            <div className="behaviour-value">

```

Comment:
can we add these components in different file, so that multiple comparisons can be avoided?

Reply:
WIP. 
Committing changes in another PR.